src/typescript/sdk/src/indexer-v2/queries/base.ts

```diff
@@ -0,0 +1,112 @@
+import "server-only";
+
+import { type Account } from "@aptos-labs/ts-sdk";
+import { type Period } from "../../const";
+import { ORDER_BY } from "../../queries";
+import { type AnyNumberString } from "../../types";
+import { TableName } from "../types/snake-case-types";
+import { postgrest } from "./client";
+import { toAccountAddressString } from "../../utils";
+
+const selectSwapsByMarketID = ({ marketID }: { marketID: AnyNumberString }) =>
+  postgrest
+    .from(TableName.SwapEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .order("market_nonce", ORDER_BY.DESC);
+
+const selectChatsByMarketID = ({ marketID }: { marketID: AnyNumberString }) =>
+  postgrest
+    .from(TableName.ChatEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .order("market_nonce", ORDER_BY.DESC);
+
+const selectPeriodicEventsByPeriod = ({
+  marketID,
+  period,
+  startTime,
+}: {
+  marketID: AnyNumberString;
+  period: Period;
+  startTime: Date;
+}) =>
+  postgrest
+    .from(TableName.PeriodicStateEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .eq("period", period)
+    .gte("start_time", startTime)
+    .order("start_time", ORDER_BY.ASC);
+
+const selectMarketLatestStateEvents = ({ marketID }: { marketID?: AnyNumberString }) =>
+  (typeof marketID === "undefined"
+    ? postgrest.from(TableName.MarketLatestStateEvent).select("*")
+    : postgrest.from(TableName.MarketLatestStateEvent).select("*").eq("market_id", marketID)
+  ).order("market_nonce", ORDER_BY.DESC);
+
+const selectMarketsPostBondingCurve = () =>
+  postgrest
+    .from(TableName.MarketLatestStateEvent)
+    .select("market_id")
+    .eq("in_bonding_curve", false);
+
+const selectUniqueMarkets = () =>
+  postgrest.from(TableName.MarketLatestStateEvent).select("market_id");
+
+const selectUserLiquidityPools = ({ provider }: { provider: Account }) =>
+  postgrest
+    .from(TableName.UserLiquidityPools)
+    .select("*")
+    .eq("provider", toAccountAddressString(provider))
+    .order("market_nonce", ORDER_BY.DESC);
```

Comment:
Not sure if this order by is useful here. This would be used on the pools page, and the only sortings we user there are APR, TVL, 24H volume and all time volume.

We might need an SQL function for this that basically joins `user_liquidity_pools` and the markets table for a given address so that the result returned is only the pools that the user has participated in, and then we'll be able to sort by the previously mentioned characteristics on that.

Reply:
1. Essentially we need to join the user liquidity pools  with the pool data
2. Indicate in `user_liquidity_pools` whether or not the user has any lp coins- check the writeset for the balance of `lp_coin_amount` at the end of the transaction
3. Consider storing quote/lp_coin_amount in the join to indicate whether or not a user has coins